ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -606,18 +616,56 @@ private[backend] trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_
               WHERE
                 archive_events.event_offset <= $pruneUpToInclusive AND
                 archive_events.contract_id = delete_events.contract_id
-            )""",
+            )"""
+    }(connection, loggingContext)
+
+    pruneWithLogging(queryDescription = "Exercise (consuming) events pruning") {
```

Comment:
Also, if this bug was not caught by a test. Consider, before fixing it, what test you can add that would have caught it.

Reply:
Yes, you're right, thanks! When not pruning all divulgence events, this will leave orphans. Oversaw this. Will amend. Wrt to the test, the new tests are only asserting prune with `prune_all_divulged_contracts`.

Reply:
Added a new test in `ParticipantPruningIT` (`PRDivulgenceArchivalPruning`) to fill this gap.